main/src/library/Array.flix

```diff
@@ -1485,7 +1485,7 @@ namespace Array {
     /// Apply the effectful function `f` to all the elements in the array `a`.
     ///
     @Time(time(f) * length(a)) @Space(space(f))
-    pub def foreach(f: a ~> Unit, a: Array[a]): Unit & Impure =
+    pub def foreach(f: a -> Unit & ef, a: Array[a]): Unit & ef =
```

Comment:
f should have effect `ef` but the whole function should be impure (because it touches an array).